pkg/presto/util.go

```diff
@@ -77,6 +77,28 @@ func DropView(queryer db.Queryer, catalog, schema, viewName string, ignoreNotExi
 	return err
 }
 
+// QueryMetadata executes a "DESCRIBE" Presto query against an existing, fully-qualified
+// table name to determine that table's column information.
+func QueryMetadata(queryer db.Queryer, catalog, schema, tableName string) ([]Column, error) {
+	rows, err := ExecuteSelect(queryer, fmt.Sprintf("DESCRIBE %s", FullyQualifiedTableName(catalog, schema, tableName)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to query the %s Presto table's metadata: %v", tableName, err)
+	}
+
+	var cols []Column
+
+	for _, row := range rows {
```

Comment:
What if code in here errors? We're still returning nil for the error.

Reply:
I think that's a fair point, but I can't see a reason where an error would occur. Type assertion in Go asserts that the row["Column"] element is not nil, and the value is of type string.

Reply:
Actually I believe you're right as that implies that if either of those checks fails - it panics which is not ideal.